### ISSUE TITLE
keep the bin/markdown.dart entrypoint?

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/markdown#248]
I'm curious how valuable the https://github.com/dart-lang/markdown/blob/master/bin/markdown.dart entrypoint is. Having it means that all packages and tools that depend on package:markdown need to spend time precompiling the entrypoint:

```
Resolving dependencies... 
Got dependencies!
Precompiling executables... 
Precompiled markdown:markdown.

Precompiled test:test.
```

**Comment by srawlins** [comment]
Hmm, I suppose as useful as other bin/ binaries that packages provide?

I use it any time I develop markdown. Users who activate it use the binary script.

I don't understand why package **a**'s executables are precompiled just because package **a** is listed in the _dependencies_ of package **b**. If developers of **b** need **a**'s executables, it would be in dev_dependencies. If package **c** has **b** as a dependency, then when a user/developer runs `pub get` in **c**, then the _dependencies_ of **b** are fetched, including **a**, but **a**'s executables are not accessible to **b** or **c**, right? What scenario require's **a**'s executables to be precompiled?

As a sibling issue, I've opened https://github.com/dart-lang/sdk/issues/34837 proposing that bin dependencies be split out.

**Comment by devoncarew** [comment]
Yeah, I'm unfamiliar with how often it's used or what the use case is...

> I use it any time I develop markdown.

That could be a script in `tools/`?

> Users who activate it use the binary script.

Is that common? Uncommon, but important when needed?

**Comment by srawlins** [comment]
> > I use it any time I develop markdown.

> That could be a script in tools/?

It could.

> > Users who activate it use the binary script.

> Is that common? Uncommon, but important when needed?

A good question. Not sure how to get `pub activate` statistics... It's just that... its a part of the package. What would it mean if I answered "uncommon but important" here?
